server/src/lib/model/db/migrate-demographics.ts

```diff
@@ -0,0 +1,26 @@
+import { getMySQLInstance } from './mysql';
+
+const db = getMySQLInstance();
+
+export async function migrateDemographics() {
+  await db.query(
+    `
+      INSERT INTO demographics (client_id, age, gender)  (
+        SELECT client_id, age, gender
+        FROM user_clients
+      )
```

Comment:
Hmm, good point; I’ve added a `FIXME` for now. I want to ensure that the join logic is sound everywhere in case a record isn’t found. I can also see problems with our current `SELECT id FROM demographics WHERE client_id = ? ORDER BY updated_at DESC LIMIT 1` solution, both for folks who don’t have a `demographics` record _and_ for folks who later add one but don’t want it to apply to their prior activity.

Should be easy accommodations to make, and I agree that it would be nice to avoid useless records where possible.

Reply:
Hmm, I'm not sure I understand what you mean about the select problems.

Reply:
I just did some pretty extensive testing for the issue I mention above, so no worries! I updated the table, it's sparse now :)